src/test/java/org/opensearch/security/auditlog/compliance/RestApiComplianceAuditlogTest.java

```diff
@@ -108,21 +114,24 @@ public void testRestApiRolesDisabledGet() throws Exception {
             .build();
 
         setup(additionalSettings);
-        TestAuditlogImpl.clear();
+        Thread.sleep(1000);
 
         rh.enableHTTPClientSSL = true;
         rh.trustHTTPServerCertificate = true;
         rh.sendAdminCertificate = true;
         rh.keystore = "kirk-keystore.jks";
-        HttpResponse response = rh.executeGetRequest("_opendistro/_security/api/rolesmapping/opendistro_security_all_access?pretty");
-        Thread.sleep(1500);
-        Assert.assertEquals(HttpStatus.SC_OK, response.getStatusCode());
-        Assert.assertTrue(TestAuditlogImpl.messages.size() > 2);
-        Assert.assertTrue(TestAuditlogImpl.sb.toString().contains("audit_request_effective_user"));
-        Assert.assertTrue(TestAuditlogImpl.sb.toString().contains("COMPLIANCE_INTERNAL_CONFIG_READ"));
-        Assert.assertTrue(TestAuditlogImpl.sb.toString().contains("COMPLIANCE_INTERNAL_CONFIG_WRITE"));
-        Assert.assertTrue(TestAuditlogImpl.sb.toString().contains("UPDATE"));
-        validateMsgs(TestAuditlogImpl.messages);
+        final List<AuditMessage> messages = TestAuditlogImpl.doThenWaitForMessages(() -> {
+            HttpResponse response = rh.executeGetRequest("_opendistro/_security/api/rolesmapping/opendistro_security_all_access?pretty");
+            Assert.assertEquals(HttpStatus.SC_OK, response.getStatusCode());
+            try {
+                Thread.sleep(1500);
+            } catch (InterruptedException e) {
+                fail("Received unexpected exception");
+            }
+            Assert.assertTrue(TestAuditlogImpl.sb.toString().contains("audit_request_effective_user"));
```

Comment:
Removing any uses of `TestAuditlogImpl.sb` will help avoid concurrency issues - that string builders contains are not safe, whereas `final List<AuditMessage> messages` will never be impacted by timing issues.

Reply:
I'll remove direct usages of `TestAuditlogImpl.sb and iterate through the messages instead

Reply:
Removed direct usages